app/routes/docs.client.samples.md

```diff
@@ -0,0 +1,160 @@
+---
+handle:
+  breadcrumb: Code Samples
+---
+
+# Code Samples
+
+Here is a collection of functions and example code that may be useful
+when working with GCN Notices. These include conversions from
+text and VOEvent alert formats to JSON and XML (for VOEvent alerts), how to save alerts,
+and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+For more examples, or to contribute your own ideas, see our [Utility Samples](https://github.com/nasa-gcn/gcn.nasa.gov/blob/CodeSamples/app/routes/docs.client.samples.md) and the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+## Parsing
+
+Within your consumer loop, use the following functions to convert the
+content of <code>message.value()</code> into other data types.
+
+The `xmltodict` package is not part of the Python standard library. You must install it by running:
+pip:
+
+```sh
+pip install xmltodict
+```
+
+```python
+
+import email
+import xml.etree.ElementTree as ET
+import xmltodict
+
+def parse_text_alert_to_dict(message_value):
+    return dict(email.message_from_bytes(message_value))
+
+def parse_voevent_alert_to_xml_root(message_value):
+    return ET.fromstring(message_value)
+
+def parse_voevent_alert_to_dict(message_value):
+    return xmltodict.parse(message_value)
+
+```
+
+## Saving
+
+Use the following to save the data to a local file:
+
+```python
+import json
+
+def save_alert(message):
+    with open('path/to/your/file', 'wb') as file:
+        file.write(message.value())
+
+```
+
+## Keep track of the last read message when restarting a client
+
+Kafka can keep track of which messages your client has read, allowing your client to recover
+missed messages after a restart by beginning at the earliest unread message
+rather than the next available message from the stream.
+
+To enable this feature, you will need to set a client Group ID using
+the configuration dictionary argument for the Consumer and change
+the auto offset reset option to the 'earliest' setting. Once this is done,
+every new client with the given Group ID will begin reading the specified topic
+at the earliest unread message.
+
+When doing this, it is recommended to turn OFF the auto commit
+feature because it can lose track of the last read message if the client
+crashes before the auto commit interval (5 seconds by default) occurs.
+Manually committing messages (i.e. storing the state of the last read message)
+once they are read is the most robust method for tracking the last read
+message.
+
+```python
+from gcn_kafka import Consumer
+
+config = {'group.id': 'my group name',
+          'auto.offset.reset': 'earliest',
+          'enable.auto.commit': False}
+
+consumer = Consumer(config=config,
+client_id='fill me in',
+client_secret='fill me in',
+domain='gcn.nasa.gov')
```

Comment:
Fix indentation.

Reply:
This and a few other comments are not resolved. Did you forget to push something?